### PR TITLE
Add basic types and image product

### PR DESCRIPTION
**Pull request by dsavchenko**

### CODE REVIEW

**Comment by dsavchenko** [comment]
@volodymyrss do we really need these new strict checks for type annotations (see failed action)?

**Comment by burnout87** [comment]
> @volodymyrss do we really need these new strict checks for type annotations (see failed action)?

I had to address them too here: https://github.com/oda-hub/oda_api/pull/163#issuecomment-1311365481

**Comment by dsavchenko** [comment]
> I had to address them too here: [#163 (comment)](https://github.com/oda-hub/oda_api/pull/163#issuecomment-1311365481)

Great. I temporarily switched off this check in this PR and will profit from your work afterwards :)

File: oda_api/data_products.py [hunk 1]
```diff
@@ -757,3 +765,84 @@ def from_arrays(cls,
                                  hdu_type = 'bintable',
                                  name = 'LIGHTCURVE')],
                    name = name)            
+
+class BinaryImageProduct:
```
Comment (volodymyrss) [inline oda_api/data_products.py:769]: so, PictureProduct, right?

Comment (dsavchenko) [reply oda_api/data_products.py]: Sorry, I committed but forgot to push

**Review by volodymyrss — APPROVED** [review]

**Comment by dsavchenko** [comment]
So, waiting for @burnout87 review?

**Review by burnout87 — APPROVED** [review]

**Comment by volodymyrss** [comment]
> It will need help though, as agreed in the example. Could you maybe just reference it here for the record?

Is this the example, @dsavchenko  https://renkulab.io/gitlab/dsavchenko/mmoda-nb2workflow-example ?

**Comment by dsavchenko** [comment]
> Is this the example, @dsavchenko https://renkulab.io/gitlab/dsavchenko/mmoda-nb2workflow-example ?

Yes

**Comment by dsavchenko** [comment]
BTW @volodymyrss do you bump the version of oda_api somehow automatically?

**Comment by volodymyrss** [comment]
> BTW @volodymyrss do you bump the version of oda_api somehow automatically?

Sort of.

**Pull request merged**